accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,106 @@
+
+- The user can see from Uyuni/SUSE Manager if a node has packages updates and patch available, along with all the information that a Salt minion is presenting when registered to Uyuni/SUSE Manager (e.g. CVE auditing)
+- Another benefit is that the user can associate staged channels to the nodes registered (along with an activation key) to associate channels to the nodes of the cluster.
+
+The agent `skuba-update`, running on each node of the cluster, automatically picks up the packages from the associated channels and proceed to automatically update the nodes.
+NOTE: `skuba-update` is **NOT** invoked by Uyuni/SUSE Manager and it is running autonomously on the nodes of the CaaS Platform cluster.
+
+For example, an organization could have a "staging" cluster with a set of staging channels. The operator would cherry-pick Kubernetes updates from the SCC channels into the staging ones and then wait for skuba-update to roll them out into the staging cluster. At a later time, once testing is green inside of the staging cluster, the packages can be promoted to the "production channels" and be rolled out to the production cluster.
+
+It is paramount to state in the documentation that the information presented for the minion is in read-only mode. In the initial iteration, Uyuni/SUSE Manager must point out in the UI that executing actions (package install, upgrade, reboot) might break the cluster if the action are run on a node of the cluster.
+If a user reboots the node, installs a patch or installs a package related to Kubernetes (`kubernetes-kubeadm kubernetes-kubelet kubernetes-client cri-o cni-plugins`) the Kubernetes may break.
+
+NOTE: both of the above actions will not cluster-coordinated, so it is entirely possible that a user assigns different channels, install packages or reboots nodes in a subset of nodes belonging to the same cluster.
+
+## Update the current Kubernetes integration
+
+Under Virtual Host Manager, there is an already Kubernetes integration that supports CaaS Platform version 3.
+The current implementation breaks with CaaS Platform version 4: the Salt runner needs to take into account that CaaS Platform version 4 switched from Docker to CRI-O.
+Uyuni/SUSE Manager must check if the underlying container engine is running Docker (and use the existing Kubernetes integration) or CRI-O (and adapt the current implementation to support CRI-O).
+The Kubernetes implementation that this RFC is targeting is the one in CaaS Platform version 4: 1.16. Other Kubernetes distributions will not be targeted, at least initially.
+
+Into the Virtual Host Manager, when a Kubernetes cluster is selected, the properties and the names of the nodes are shown, together with CPU Arch, CPU Sockets and RAM of the node.
+
+After the user has imported the `kubeconfig`, obtained from CaaS Platform, the process would be:
+
+- When a user registers a new cluster, we introduce a new workflow to bootstrap (using Systems > Bootstrapping) each node of the cluster
```

Comment:
What do you mean with "register a new cluster"? I thought that was the result of bootstraping each node of the cluster

Reply:
I `s/register/defines` - would that clarify?

Reply:
Yep, that would clarify. So the steps would be:

1. User logs into SUMA, defines a new cluster by clicking some button
1. User creates the nodes using autoyast/terraform. Nodes trigger somehow the bootstrap script from SUMA
1. Profit!

Is that correct?

Reply:
The point 2 is not correct: the user creates the nodes using AutoYaST in Uyuni/SUSE Manager (following the documentation, see "AutoYaST Deployment documentation" in this RFC) or using Terraform (outside the scope of Uyuni/SUSE Manager).
When the nodes of the cluster have been bootstrapped (in the CaaSP domain), then the user imports the `kubeconfig` into SUSE Manager: the new workflow will bootstrap (in the Uyuni/SUSE Manager domain, read: register) the nodes into Uyuni/SUSE Manager.

Reply:
Pushed clarification in this sentence.

Reply:
Please start with the most simple case: An exisiting CaaSP cluster is added to SUSE Manager in order to control updates/patches visible to the nodes.
Deployment of a cluster (or new nodes) should be the second iteration of this feature.